cmd/kubectl-kubestellar/cmd/ensure/location.go

```diff
@@ -0,0 +1,166 @@
+			// want the help to be displayed when the error is due to an
+			// invalid command.
+			cmd.SilenceUsage = true
+			err := ensureLocation(cmd, args, cliOpts)
+			return err
+		},
+	}
+
+	// Add flag for IMW name
+	cmdLocation.Flags().StringVar(&imw, "imw", "", "IMW name")
+	cmdLocation.MarkFlagRequired("imw")
+	return cmdLocation
+}
+
+// The IMW name is provided by the --imw flag (stored in the "imw" string
+// variable), and the location name is a command line argument.
+// Labels to check are provided as additional arguments in key=value pairs.
+// In this function we will:
+// - Work in the provided IMW
+// - Check if APIBinding "edge.kubestellar.io" exists in IMW, create if not
+// - Check for SyncTarget of provided name in IMW, create if not
+// - Check that SyncTarget has an "id" label matching the Location name
+// - Ensure that SyncTarget has the labels provided by the user
+// - Check for Location of provided name in IMW, create if not
+// - Ensure that Location has the labels provided by the user
+// - If Location "default" exists, delete it
+func ensureLocation(cmdLocation *cobra.Command, args []string, cliOpts *genericclioptions.ConfigFlags) error {
+	locationName := args[0]
+	labels := args[1:]
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+
+	// Print all flags and their values if verbosity level is at least 1
+	cmdLocation.Flags().VisitAll(func(flg *pflag.Flag) {
+		logger.V(1).Info(fmt.Sprintf("Command line flag %s=%s", flg.Name, flg.Value))
+	})
+
+	// Make sure user provided location name is valid
+	err := plugin.CheckLocationName(locationName)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Problem with location name %s", locationName))
+		return err
+	}
+
+	// Make sure user provided labels are valid
+	err = plugin.CheckLabelArgs(labels)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Problem with label arguments %s", labels))
+		return err
+	}
+
+	// Set context to root, later on we will append the IMW name to the root server
+	configContext := "root"
+	cliOpts.Context = &configContext
+
+	// Get client config from flags
+	config, err := cliOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "Failed to get config from flags")
+		return err
+	}
+
+	// Update host to work on objects within IMW
+	config.Host += ":" + imw
+	logger.V(1).Info(fmt.Sprintf("Set host to %s", config.Host))
+
+	// Create client-go instance from config
+	kcpClient, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		logger.Error(err, "Failed create client-go instance")
+		return err
+	}
+
+	// Check that APIBinding exists, create if not
+	// This function prints its own log messages, so no need to add any here.
+	err = plugin.VerifyOrCreateAPIBinding(kcpClient, ctx, "edge.kubestellar.io", "edge.kubestellar.io", "root:espw")
+	if err != nil {
+		return err
+	}
+
+	// Create client-go instance from config
+	client, err := clientset.NewForConfig(config)
```

Comment:
I would qualify `client` and `clientset` here, to make the complementary to the `kcp` ones above; perhaps with `k8s`?